### PR TITLE
warn when memory usage is high on the GPU

### PR DESCRIPTION
**Pull request by matthiasdiener**
**Questions for the review**:
- [ ] Is the scope and purpose of the PR clear?
  - [ ] The PR should have a description.
  - [ ] The PR should have a guide if needed (e.g., an ordering).
- [ ] Is every top-level method and class documented? Are things that should be documented actually so?
- [ ] Is the interface understandable? (I.e. can someone figure out what stuff does?) Is it well-defined?
- [ ] Does the implementation do what the docstring claims?
- [ ] Is everything that is implemented covered by tests?
- [ ] Do you see any immediate risks or performance disadvantages with the design? Example: what do interface normals attach to?

### CODE REVIEW

**Comment by matthiasdiener** [comment]
This is a relatively crude way of checking (in particular, it doesn't capture if there are temporary allocations within the time step that exceed the memory size), but I couldn't come up with a better way to do this.

File: mirgecom/logging_quantities.py [hunk 1]
```diff
@@ -428,6 +428,13 @@ def __call__(self) -> Optional[float]:
             warn(f"cudaMemGetInfo failed with error {ret}.")
             return None
         else:
+            if self.free.value / self.total.value < 0.1:
```
Comment (inducer) [inline mirgecom/logging_quantities.py:431]: It doesn't belong here, but :shrug: IDK.

Comment (inducer) [inline mirgecom/logging_quantities.py:431]: While I'm at it, why are `self.free` and `self.total` instance attributes? I don't see a need for them to be long-lived.

**Review by MTCam — APPROVED** [review]
In general, I like this warning.   Not intending to subvert @inducer's concerns.  I don't see anything that should prevent merge.

File: mirgecom/logging_quantities.py (hunk 1 above)
Comment (matthiasdiener) [reply mirgecom/logging_quantities.py]: That's a good catch, thanks! Unfortunately, I can't remember why I wrote the code this way in #255 (possibly to reduce overhead from the `ctypes.c_size_t()` calls?). I can't see a performance difference with [844db39](https://github.com/illinois-ceesd/mirgecom/pull/874/commits/844db3950e53f4a04fdd24b6363b2d447de2d348), so it should be ok to change these to local variables.

**Comment by inducer** [comment]
Don't let me hold up the merge here, I think this is fair enough.

**Pull request merged**